pym/bob/cmds/misc.py

```diff
@@ -18,6 +18,22 @@
 import argparse
 import sys
 
+
+def doProject(argv, bobRoot):
+    def createProject(packages):
+        print ("you need to register a plugin.")
+
+    parser = argparse.ArgumentParser(prog="bob project", description='Generate Project Files')
+    parser.add_argument('project', type=str, nargs=1,
+                        help="plugin to use for project file generation")
```

Comment:
You should get the package name here. Anything after that can go to the plugin:

```python
    parser.add_argument('packages', metavar='PACKAGE',
        help="(Sub-)package to build")
    parser.add_argument('args', nargs=argparse.REMAINDER,
                        help="Arguments for project generator")
args = parser.parse_args(argv)
```

Then pass `args.args` to the plugin.